#include "signup.h"
#include "ui_signup.h"
#include <QRegularExpression>
#include <QRegularExpressionValidator>
signup::signup(QWidget *parent) : QDialog(parent), ui(new Ui::signup) {
    ui->setupUi(this);
    setWindowModality(Qt::ApplicationModal);  // ì „ì²´ ì•± ëª¨ë‹¬
    setWindowFlag(Qt::WindowStaysOnTopHint);  // ìœ„ë¡œ ì˜¤ê¸°
    setWindowFlag(Qt::Dialog);                // ëŒ€í™”ìƒìž ìŠ¤íƒ€ì¼


    ui->pw_signup->setEchoMode(QLineEdit::Password);


    QRegularExpression re("^[0-9]*$");
    ui->accnum_signup->setValidator(new QRegularExpressionValidator(re, this));
}


signup::~signup() {
    delete ui;
}


void signup::on_signup_btn_clicked() {
    QString name = ui->name_signup->text();
    QString acc = ui->accnum_signup->text();
    QString id = ui->id_signup->text();
    QString pw = ui->pw_signup->text();

    if (name.isEmpty() || acc.isEmpty() || id.isEmpty() || pw.isEmpty()) {
        QMessageBox::warning(this, "ì˜¤ë¥˜", "ëª¨ë“  ì •ë³´ë¥¼ ìž…ë ¥í•˜ì„¸ìš”.");
        return;
    }

    // ðŸ”¹ 1. ID ì¤‘ë³µê²€ì‚¬
    QSqlQuery checkQuery;
    checkQuery.prepare("SELECT id FROM User WHERE id=?");
    checkQuery.addBindValue(id);
    checkQuery.exec();

    if (checkQuery.next()) {
        QMessageBox::warning(this, "ì¤‘ë³µ", "ì´ë¯¸ ì¡´ìž¬í•˜ëŠ” IDìž…ë‹ˆë‹¤.");
        return;
    }

    // ðŸ”¹ 2. ACC(ê³„ì¢Œë²ˆí˜¸) ì¤‘ë³µê²€ì‚¬
    QSqlQuery accQuery;
    accQuery.prepare("SELECT acc FROM users WHERE acc=?");
    accQuery.addBindValue(acc);
    accQuery.exec();

    if (accQuery.next()) {
        QMessageBox::warning(this, "ì¤‘ë³µ", "ì´ë¯¸ ì¡´ìž¬í•˜ëŠ” ê³„ì¢Œë²ˆí˜¸ìž…ë‹ˆë‹¤.");
        return;
    }

    // ðŸ”¹ 3. DB ì‚½ìž…
    QSqlQuery query;
    query.prepare("INSERT INTO User (id, pw, name, acc) VALUES (?,?,?,?)");
    query.addBindValue(id);
    query.addBindValue(pw);
    query.addBindValue(name);
    query.addBindValue(acc);

    if (query.exec()) {
        QMessageBox::information(this, "ì™„ë£Œ", "íšŒì›ê°€ìž…ì´ ì™„ë£Œë˜ì—ˆìŠµë‹ˆë‹¤!");
        emit gotoLogin();
    } else {
        QMessageBox::warning(this, "ì‹¤íŒ¨", "DB ì €ìž¥ ì˜¤ë¥˜ê°€ ë°œìƒí–ˆìŠµë‹ˆë‹¤.");
    }
}



void signup::on_signupPwShow_btn_clicked() {
    bool isPassword = (ui->pw_signup->echoMode() == QLineEdit::Password);
    ui->pw_signup->setEchoMode(isPassword ? QLineEdit::Normal : QLineEdit::Password);
}
